docs/src/further_topics/ugrid/operations.rst

```diff
@@ -0,0 +1,953 @@
+    * - `Equality`_
+      - |tagline: equality|
+    * - `Combining Cubes`_
+      - |tagline: combining cubes|
+    * - `Arithmetic`_
+      - |tagline: arithmetic|
+
+..
+    Below: use demo code over prose wherever workable. Headings aren't an
+     exhaustive list (can you think of any other popular operations?).
+
+Making a Mesh
+-------------
+.. |tagline: making a mesh| replace:: |new|
+
+.. rubric:: |tagline: making a mesh|
+
+**Already have a file?** Consider skipping to `Load`_.
+
+Creating Iris objects from scratch is a highly useful skill for testing code
+and improving understanding of how Iris works. This knowledge will likely prove
+particularly useful when converting data into the Iris mesh data model from
+structured formats and non-UGRID mesh formats.
+
+The objects created in this example will be used where possible in the
+subsequent example operations on this page.
+
+.. dropdown:: :opticon:`code`
+
+    .. doctest:: ugrid_operations
+
+        >>> import numpy as np
+
+        >>> from iris.coords import AuxCoord
+        >>> from iris.experimental.ugrid import Connectivity, Mesh
+
+        # Going to create the following mesh
+        #  (node indices are shown to aid understanding):
+        #
+        #  0----1
+        #  |    |\
+        #  | +  |+\
+        #  2----3--4
+
+        >>> node_x = AuxCoord(
+        ...     points=[0.0, 5.0, 0.0, 5.0, 8.0],
+        ...     standard_name="longitude",
+        ...     units="degrees_east",
+        ...     long_name="node_x_coordinates",
+        ... )
+        >>> node_y = AuxCoord(points=[3.0, 3.0, 0.0, 0.0, 0.0], standard_name="latitude")
+
+        >>> face_x = AuxCoord([2.0, 6.0], "longitude")
+        >>> face_y = AuxCoord([1.0, 1.0], "latitude")
+
+        >>> edge_node_c = Connectivity(
+        ...     indices=[[0, 1], [0, 2], [1, 3], [1, 4], [2, 3], [3, 4]],
+        ...     cf_role="edge_node_connectivity",
+        ...     attributes={"demo": "Supports every standard CF property"},
+        ... )
+
+        # Create some dead-centre edge coordinates.
+        >>> edge_x, edge_y = [
+        ...     AuxCoord(
+        ...         node_coord.points[edge_node_c.indices_by_location()].mean(axis=1),
+        ...         node_coord.standard_name,
+        ...     )
+        ...     for node_coord in (node_x, node_y)
+        ... ]
+
+        >>> face_indices = np.ma.masked_equal([[0, 1, 3, 2], [1, 4, 3, 999]], 999)
+        >>> face_node_c = Connectivity(
+        ...     indices=face_indices, cf_role="face_node_connectivity"
+        ... )
+
+        >>> my_mesh = Mesh(
+        ...     long_name="my_mesh",
+        ...     topology_dimension=2,  # Supports 2D (face) elements.
+        ...     node_coords_and_axes=[(node_x, "x"), (node_y, "y")],
+        ...     connectivities=[edge_node_c, face_node_c],
+        ...     edge_coords_and_axes=[(edge_x, "x"), (edge_y, "y")],
+        ...     face_coords_and_axes=[(face_x, "x"), (face_y, "y")],
+        ... )
```

Comment:
Is it worth also printing the Mesh out at this point ?

Reply:
d2e504f66a1f46335ae441c9494956ffb2aafb09